### PR TITLE
Added Path Alias Exists condition

### PR DESCRIPTION
**Pull request by bbujisic**
There was an error with doctrine. The new pull request fixed it.

### CODE REVIEW

File: src/Plugin/Condition/PathAliasExists.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+
+/**
+ * Provides a 'Node Type' condition.
+ *
+ * @Condition(
+ *   id = "path_alias_exists",
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: Needs a rules_ prefix.

File: src/Plugin/Condition/PathAliasExists.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *   }
+ * )
+ *
+ * @todo: Add alias context's cleaning callback information from Drupal 7.
+ * @todo: Review language context's type: token type does not exist so string is used.
+ * @todo: Add language context's options list information from Drupal 7.
+ * @todo: Add language context's default value information from Drupal 7.
+ */
+
+class PathAliasExists extends ConditionPluginBase {
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: Needs a comment.

File: src/Plugin/Condition/PathAliasExists.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    $alias = $this->getContextValue('alias');
+    $language = $this->getContextValue('language');
+
+    $path = \Drupal::service('path.alias_manager.cached')->getPathByAlias($alias, $language);
+
+    return !($path == $alias);
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: That's weird ;)  -> $path != $alias. Could use a comment also I guess.

File: src/Plugin/Condition/PathAliasExists.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    $alias = $this->getContextValue('alias');
+    $language = $this->getContextValue('language');
+
+    $path = \Drupal::service('path.alias_manager.cached')->getPathByAlias($alias, $language);
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: This should be an injected service. You can do that by implementing \Drupal\Core\Plugin\ContainerFactoryPluginInterface.

See \Drupal\action\Plugin\Action\EmailAction for an example.

File: src/Plugin/Condition/PathAliasExists.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    return !($path == $alias);
+  }
+
+}
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: Missing new line?

File: src/Plugin/Condition/PathAliasExists.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ * @file
+ * Contains \Drupal\rules\Plugin\Condition\PathAliasExists.
+ */
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+
+/**
+ * Provides a 'Node Type' condition.
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php]: Needs an update.

File: src/Plugin/Condition/PathAliasExists.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+ * )
+ *
+ * @todo: Add alias context's cleaning callback information from Drupal 7.
+ * @todo: Review language context's type: token type does not exist so string is used.
+ * @todo: Add language context's options list information from Drupal 7.
+ * @todo: Add language context's default value information from Drupal 7.
+ */
+class PathAliasExists extends ConditionPluginBase implements ContainerFactoryPluginInterface {
+
+  protected $aliasManager;
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php:44]: Misses documentation.

File: src/Plugin/Condition/PathAliasExists.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  /**
+   * Constructs a PathAliasExists object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\CacheDecorator\AliasManagerCacheDecorator $alias_manager
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php:55]: Is there an interface we can type-hint instead? Hinting on a cache decorator seems wrong.

File: src/Plugin/Condition/PathAliasExists.php [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    return t('URL alias exists');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    $alias = $this->getContextValue('alias');
+    $language = $this->getContextValue('language');
+
```
Comment (fago) [inline src/Plugin/Condition/PathAliasExists.php:85]: Unnecessary new lines?

**Comment by fubhy** [comment]
Same as in https://github.com/fago/rules/pull/6#issuecomment-44294584 

Just a side-note: I think we should squash commits in these pull requests to reduce the number of "fixing this" / "whoops, typo!" / etc. commits ending up in the rules git log. Github automatically updates the pull request if you squash your commits.

**Comment by fubhy** [comment]
Continued in https://github.com/fago/rules/pull/32

**Pull request closed without merging**